packages/kit/src/vite/index.js

```diff
@@ -76,6 +76,9 @@ export function sveltekit() {
  * @return {import('vite').Plugin}
  */
 function kit() {
+	/** @type {import('vite').UserConfig} */
+	let original_config;
```

Comment:
i'm actually surprised this works because i've run into Vite mutating the config

`original_vite_config` might be a clearer name to differentiate from the svelte config

Reply:
you're right, vite does mutate the config, or at least if other plugins mutate the config instead of returning a config object 🤔 I think it's rare enough that maybe it's fine for simplicity now. otherwise we'd have to deep clone it, or pre compute the warnings logs beforehand.

Reply:
vite config hook docs explicitly mention that plugins are allowed to mutate the config that is passed in
To get the unmodified one our best shot is reading the config again, the next best thing is a 'pre' plugin that clones it in it's own config hook. (hoping no other plugin has run before it)

Reply:
otoh even if it wasn't a user modification, the config that kit receives is the effective one, so it may still be useful to let the user know that something has a different idea. If they are using a plugin that needs a config that sveltekit does not allow the two are incompatible.

Reply:
My concern was more that we read it, then it's mutated, and we no longer know what the original config is because our version has been mutated

We can't clone the entire config because it has functions and things in it that are hard to clone, but I think the fields we need are all simple fields, so we could first filter out the fields we need and then clone those